src/connectors/assetpanda.py

```diff
@@ -0,0 +1,185 @@
+"""Asset Panda
+Collect asset information from Asset Panda using an API token
+"""
+
+from runners.helpers import db, log
+from runners.helpers.dbconfig import ROLE as SA_ROLE
+
+import requests
+from urllib.error import HTTPError
+from .utils import yaml_dump
+
+from functools import reduce
+import re
+from datetime import datetime
+
+
+PAGE_SIZE = 50
+
+CONNECTION_OPTIONS = [
+    {
+        'name': 'asset_entity_id',
+        'title': 'AssetPanda Entity ID',
+        'prompt': 'Your AssetPanda Asset Entity ID',
+        'type': 'int',
+        'required': True,
+        'secret': False
+    },
+    {
+        'name': 'token',
+        'title': 'AssetPanda API Token',
+        'prompt': 'Your AssetPanda API Token',
+        'type': 'str',
+        'secret': True
+    }
+]
+
+LANDING_TABLE_COLUMNS = [
+    ('INSERT_ID', 'NUMBER IDENTITY START 1 INCREMENT 1'),
+    ('RAW', 'VARIANT'),
+    ('ID', 'VARCHAR(256)'),
+    ('INSERT_AT', 'TIMESTAMP_LTZ(9)')
+]
+
+
+###
+# Helper Functions
+###
+
+
+def get_list_objects_and_total_from_get_object(result: dict) -> (list, int):
+    """Retrieve the values needed from the results objects"""
+    try:
+        list_object: list = result["objects"]
+        total_object_count: int = result["totals"]["objects"]
+    except BaseException:
+        raise AttributeError("error format result")
+    return (list_object, total_object_count)
+
+
+def reduce_fields(accumulators: dict, field: dict) -> str:
```

Comment:
might this be better named "accumulated_value" or "result"? it's a single value that's passed through repeated calls of the reducer.